lib/vrl/compiler/src/value/target.rs

```diff
@@ -164,14 +164,14 @@ impl Value {
                 .as_object()
                 .and_then(|map| fields.iter().find_map(|field| map.get(field.as_str()))),
             Index(index) => self.as_array().and_then(|array| {
-                let index = *index;
-
-                if index < 0 {
-                    let index = index % array.len() as i64;
-                    array.get(index.abs() as usize)
-                } else {
-                    array.get(index as usize)
+                let len = array.len() as i64;
+                if *index >= len || index.abs() - 1 >= len {
```

Comment:
Nit:
```suggestion
                if *index >= len || -index + 1 >= len {
```
would make the intention a little bit clearer imo.

Reply:
I guess it's a personal preference. I actually prefer the existing impl.

Reply:
Yep, fair 😄